packages/ui/src/theme/createTheme/createComponentCSS.ts

```diff
@@ -0,0 +1,152 @@
+    });
+
+    str += `${baseSelector} { ${propsToString(other)} }\n`;
+  }
+
+  str += addVars(baseSelector, _vars);
+
+  Object.entries(_modifiers).forEach(([key, value]) => {
+    if (value && Object.keys(value).length) {
+      str += recursiveComponentCSS(`${baseSelector}--${key}`, value);
+    }
+  });
+
+  Object.entries(_element).forEach(([key, value]) => {
+    if (value && Object.keys(value).length) {
+      str += recursiveComponentCSS(`${baseSelector}__${key}`, value);
+    }
+  });
+
+  return str;
+}
+
+/**
+ * This will take a component theme and create the appropriate CSS for it.
+ *
+ */
+export function createComponentCSS(
+  themeName: string,
+  components: Array<ComponentsTheme>,
+  tokens: WebTheme['tokens'],
+  breakpoints: DefaultTheme['breakpoints']
+) {
+  let cssText = '';
+
+  components.forEach(({ name, theme, overrides }) => {
+    const baseComponentClassName = `amplify-${name}`;
+    const componentClassName = `[data-amplify-theme="${themeName}"] .${baseComponentClassName}`;
+    // unwrap the component theme
+    // if it is a function: call it with the defaultTheme to get a static object
+    const componentTheme: BaseTheme = isFunction(theme)
+      ? (theme(tokens) as BaseTheme)
+      : theme;
+
+    cssText += recursiveComponentCSS(componentClassName, componentTheme);
+
+    // if the component theme has overrides
+    // generate the appropriate CSS for each of them
+    if (overrides) {
+      overrides.forEach((override) => {
+        // unwrap the override component theme just like above
+        const componentTheme: BaseTheme = isFunction(override.theme)
+          ? (override.theme(tokens) as BaseTheme)
+          : override.theme;
+
+        if ('mediaQuery' in override) {
+          cssText += `@media (${
+            override.mediaQuery
+          }) {\n ${recursiveComponentCSS(
+            componentClassName,
+            componentTheme
+          )} \n}`;
+        }
+        if ('breakpoint' in override) {
+          const breakpoint = breakpoints.values[override.breakpoint];
+          cssText += `\n@media (min-width: ${breakpoint}px) {\n ${recursiveComponentCSS(
+            componentClassName,
+            componentTheme
+          )} \n}`;
+        }
+        if ('selector' in override) {
+          cssText += recursiveComponentCSS(
+            `${override.selector} .${baseComponentClassName}`,
+            componentTheme
+          );
+        }
+        if ('colorMode' in override) {
+          cssText += `
+@media (prefers-color-scheme: ${override.colorMode}) {
+  ${recursiveComponentCSS(
+    `[data-amplify-theme="${themeName}"][data-amplify-color-mode="system"] .${baseComponentClassName}`,
+    componentTheme
+  )}
+}
+`;
+          cssText += recursiveComponentCSS(
+            `[data-amplify-theme="${themeName}"][data-amplify-color-mode="${override.colorMode}"] .${baseComponentClassName}`,
+            componentTheme
+          );
+        }
```

Comment:
Nit: would this formatting be more in line with the rest of the statements in this file?

```suggestion
        if ('colorMode' in override) {
          cssText += `@media (prefers-color-scheme: ${override.colorMode}) {
            ${recursiveComponentCSS(
              `[data-amplify-theme="${themeName}"][data-amplify-color-mode="system"] .${baseComponentClassName}`,
              componentTheme
            )}
          }`;
          cssText += recursiveComponentCSS(
            `[data-amplify-theme="${themeName}"][data-amplify-color-mode="${override.colorMode}"] .${baseComponentClassName}`,
            componentTheme
          );
        }
```